### PR TITLE
[MOSIP-29852]Update InfraProviderServiceImpl.java

### PR DESCRIPTION
**Pull request by balaji-alluru**

### CODE REVIEW

File: partner/partner-management-service/src/main/java/io/mosip/pms/partner/misp/service/impl/InfraProviderServiceImpl.java [hunk 1]
```diff
@@ -274,7 +274,30 @@ public MISPLicenseResponseDto regenerateKey(String mispId) {
 				response.setLicenseKeyExpiry(newLicenseKey.getValidToDate());
 				response.setLicenseKeyStatus("Active");
 				response.setProviderId(mispId);
-				notify(MapperUtils.mapDataToPublishDto(newLicenseKey), EventType.MISP_LICENSE_UPDATED);
+				List<PartnerPolicyRequest> approvedPolicyMappedReq = partnerPolicyRequestRepository.findByPartnerId(mispId);
```
Comment (ase-101) [inline partner/partner-management-service/src/main/java/io/mosip/pms/partner/misp/service/impl/InfraProviderServiceImpl.java:277]: Seems the whole logic should be corrected.
One MISP partner can have any number of License key created. MISP license key should NOT be bound to policy.
When the regenerate request is received, we check only for partner existence and its status and then generate the License Key. and also raise an event to IDA with MISP partner details and the new license key.

Comment (balaji-alluru) [reply partner/partner-management-service/src/main/java/io/mosip/pms/partner/misp/service/impl/InfraProviderServiceImpl.java]: Updated the code as suggested anusha

File: partner/partner-management-service/src/main/java/io/mosip/pms/partner/misp/service/impl/InfraProviderServiceImpl.java [hunk 2]
```diff
@@ -259,51 +260,65 @@ public MISPLicenseResponseDto regenerateKey(String mispId) {
...
 					MISPErrorMessages.MISP_IS_INACTIVE.getErrorMessage());
 		}
-		List<MISPLicenseEntity> mispLicenses = mispLicenseRepository.findByMispId(mispId);
-		boolean isActiveLicenseExists = false;
+		List<MISPLicenseEntity> mispValidLicenses = mispLicenseRepository.findByMispIdandExpirydate(mispId);
+		if(mispLicenseRepository.findByMispId(mispId).isEmpty()) {
+			throw new MISPServiceException(MISPErrorMessages.MISP_LICENSE_KEY_NOT_ASSOCIATED_MISP_ID.getErrorCode(),
+					MISPErrorMessages.MISP_LICENSE_KEY_NOT_ASSOCIATED_MISP_ID.getErrorMessage());
+		}
+		List<PartnerPolicyRequest> approvedPolicyMappedReq = partnerPolicyRequestRepository.findByPartnerId(mispId);
```
Comment (ase-101) [inline partner/partner-management-service/src/main/java/io/mosip/pms/partner/misp/service/impl/InfraProviderServiceImpl.java:268]: Does this query return ordered entries? When an existing MISP partner is assigned with a new policy how does this behave? Will this query always return the latest mapped policy?

Comment (balaji-alluru) [reply partner/partner-management-service/src/main/java/io/mosip/pms/partner/misp/service/impl/InfraProviderServiceImpl.java:268]: as of now, we are expecting only 1 policy mapping has to be present for misp partner. For the issues in this regard, we already got a story created and gave it to BA to comeup with the approach on how to fix it.

File: partner/partner-management-service/src/main/java/io/mosip/pms/partner/misp/service/impl/InfraProviderServiceImpl.java [hunk 3]
```diff
@@ -259,51 +260,65 @@ public MISPLicenseResponseDto regenerateKey(String mispId) {
...
+		String policyId = (!approvedPolicyMappedReq.isEmpty() && !approvedPolicyMappedReq.get(0).getPolicyId().isBlank())?approvedPolicyMappedReq.get(0).getPolicyId():null;
```
[the remark this answers is not in the document]
Comment (balaji-alluru) [reply partner/partner-management-service/src/main/java/io/mosip/pms/partner/misp/service/impl/InfraProviderServiceImpl.java]: Updated the same as suggested

**Review by LoganathanSekar7627 — APPROVED** [review]

**Review by ckm007 — APPROVED** [review]

**Pull request merged**